Paralogues.pm

```diff
@@ -0,0 +1,705 @@
+      'start',
+      'end',
+      'strand',
+      'alleles',
+      'clinical_significance',
+      'source',
+      'consequence'
+    );
+  }
+  my $source = defined $self->{vcf} ? basename $self->{vcf} : 'Ensembl API'; 
+
+  my $fields = join(':', @fields);
+  my $description = "Variants from $source in paralogue proteins (colon-separated fields: $fields)";
+  return { PARALOGUE_VARIANTS => $description };
+}
+
+sub _join_results {
+  my $self = shift;
+  my $all_results = shift;
+  my $res = shift;
+
+  if ($self->{config}->{output_format} eq 'json' || $self->{config}->{rest}) {
+    # Group results for JSON and REST
+    $all_results->{"Paralogues"} = [] unless defined $all_results->{"Paralogues"};
+    push(@{ $all_results->{"Paralogues"} }, $res);
+  } else {
+    # Create array of results per key
+    for (keys %$res) {
+      $all_results->{$_} = [] if !$all_results->{$_};
+      push(@{ $all_results->{$_} }, $res->{$_} || "NA");
+    }
+  }
+  return $all_results;
+}
+
+sub _summarise_vf {
+  my $vf = shift;
+  my $var = join(':', (
+    $vf->name,
+    $vf->seq_region_name,
+    $vf->seq_region_start,
+    $vf->seq_region_end,
+    $vf->strand,
+    $vf->allele_string,
+    join('/', @{$vf->get_all_clinical_significance_states}),
+    $vf->source_name,
+    $vf->display_consequence,
+  ));
+  return { PARALOGUE_VARIANTS => $var };
+}
+
+sub run {
+  my ($self, $tva) = @_;
+
+  my $vf = $tva->variation_feature;
+  my $translation_start = $tva->base_variation_feature_overlap->translation_start;
+  return {} unless defined $translation_start;
+
+  my $homologies = [];
+  if ($self->{remote}) {
+    $homologies = $self->_get_database_homologies($tva->transcript);
+  } else {
+    my $translation = $tva->transcript->translation;
+    $homologies = $self->{_cache_homologies}->{$translation->stable_id} ||=
+      $self->_get_paralogues($vf, $translation);
+  }
+  return {} unless @$homologies;
+
+  my $all_results = {};
+  for my $aln (@$homologies) {
+    $aln = $aln->get_SimpleAlign if $aln->isa('Bio::EnsEMBL::Compara::Homology');
+    next unless $aln->isa('Bio::SimpleAlign');
+
+    my ($chr, $start, $end) = $self->_get_paralogue_coords($tva, $aln);
+    next unless defined $chr and defined $start and defined $end;
+
+    if (defined $self->{vcf}) {
+      # get variants from custom VCF file
+      my @data = @{$self->get_data($chr, $start - 2, $end)};
+      $all_results = $self->_join_results($all_results, $_) for @data;
+    } else {
+      # get Ensembl variants from mapped genomic coordinates
+      my $config  = $self->{config};
+      my $reg     = $config->{reg};
+      my $species = $config->{species};
+
+      $self->{sa}  ||= $reg->get_adaptor($species, 'core', 'slice');
+      $self->{vfa} ||= $reg->get_adaptor($species, 'variation', 'variationfeature');
+
+      my $slice = $self->{sa}->fetch_by_region('chromosome', $chr, $start, $end);
+      foreach my $var ( @{ $self->{vfa}->fetch_all_by_Slice($slice) } ) {
```

Comment:
You could check if the slice is defined before fetching by slice.

Reply:
Added, thanks!